x-pack/plugins/cases/server/client/cases/delete.ts

```diff
@@ -88,38 +94,46 @@ export async function deleteCases(ids: string[], clientArgs: CasesClientArgs): P
       entities: Array.from(entities.values()),
     });
 
-    await Promise.all(
-      ids.map((id) =>
-        caseService.deleteCase({
-          unsecuredSavedObjectsClient,
-          id,
-        })
-      )
-    );
+    const deleteCasesMapper = async (id: string) =>
+      caseService.deleteCase({
+        unsecuredSavedObjectsClient,
+        id,
+      });
 
-    const comments = await Promise.all(
-      ids.map((id) =>
-        caseService.getAllCaseComments({
+    // Ensuring we don't too many concurrent deletions running.
+    await pMap(ids, deleteCasesMapper, {
+      concurrency: MAX_CONCURRENT_SEARCHES,
+    });
+
+    const getCommentsMapper = async (id: string) =>
+      caseService.getAllCaseComments({
+        unsecuredSavedObjectsClient,
+        id,
+      });
+
+    // Ensuring we don't too many concurrent get running.
+    const comments = await pMap(ids, getCommentsMapper, {
```

Comment:
nit: This isn't a change you made but if I understand the previous code correctly `comments` will be an array of saved object find responses (essentially an array of arrays `[[1,2,3],[4,5,6]]`). If we flattened it I wonder if it would make the `deleteCommentsMapper` a little easier to understand 🤷‍♂️ . That way we could avoid the `pMap` inside of the mapper.

Reply:
That's smart! As we gonna use `search_after` to iterate over all objects I will leave it for another PR and create a ticket to track it.